packages/messages/src/use-cases/list-user-chats.js

```diff
@@ -0,0 +1,131 @@
+// Internal dependencies
+const logger = require('../utils/logger');
+
+/**
+ *
+ * @param {import('mongoose').Model} chatModel Chat model
+ * @param {import('mongoose').Model} messageModel Message model
+ * @returns {
+ *  ({ user, page }: { user: { _id: string }; page: number }) => Promise<{
+ *    pages: number;
+ *    result: unknown[]
+ *  }>
+ * }
+ */
+const listUserChats = (chatModel, messageModel) => async ({
+  user,
+  page,
+}) => {
+  // Validate user
+  if (!Object.prototype.hasOwnProperty.call(user, '_id')) {
+    const errorMessage = 'User does not contain the property "_id"';
+
+    logger.error(`[messages]: ${errorMessage}`);
+    throw new Error(errorMessage);
+  }
+
+  // Validate page
+  if (!Number.isInteger(page) || page < 1) {
+    const errorMessage = 'Page must be a positive integer greater than 0';
+
+    logger.error(`[messages]: ${errorMessage}`);
+    throw new Error(errorMessage);
+  }
+
+  logger.info(`[messages]: Listing chats for user ${user._id}`);
+
+  let chats = [];
+  let pages = 0;
+
+  try {
+    /** @type {import('mongoose').FilterQuery<unknown>} */
+    const query = {
+      $or: [{ hostId: user._id }, { customerId: user._id }],
+      deletedAt: null,
+    };
+
+    /** @type {import('mongoose').QueryOptions} */
+    const options = {
+      skip: (page - 1) * 10,
+      limit: 10,
+      rawResult: true,
+    };
+
+    /** @type {{[key: string]: 0 | 1}} */
+    const fields = {
+      _id: 1,
+      bookingId: 1,
+      hostId: 1,
+      customerId: 1,
+      createdAt: 1,
+      updatedAt: 1,
+      deletedAt: 0,
+    };
+
+    // Find with pagination
+    const count = await chatModel.countDocuments(query);
+    const response = await chatModel.find(query, fields, options);
+
+    if (response) {
+      /** @type {{ [id: string]: number}} */
+      const chatMapIds = {};
+
+      const messagesAsyncQueue = [];
+
+      /** @type {{[key: string]: 0 | 1}} */
+      const messageFields = {
+        _id: 1,
+        chatId: 1,
+        text: 1,
+        createdAt: 1,
+        deletedAt: 0,
+        createdBy: 1,
+      };
+
+      /** @type {import('mongoose').QueryOptions} */
+      const messageOptions = {
+        sort: { createdAt: -1 },
+        rawResult: true,
+      };
+
+      for (let i = 0; i < response.length; i += 1) {
+        chatMapIds[response[i]._id] = i;
+        response[i].lastMessage = null;
+
+        messagesAsyncQueue.push(
+          messageModel.findOne(
+            {
+              chatId: response[i]._id,
+              deletedAt: null,
+            },
+            messageFields,
+            messageOptions,
+          ),
+        );
+      }
```

Comment:
@JoseNoriegaa I'm not sure why are we doing this? because this will break with the number of messages is too big making the array overflow throwing an exception. Why don't we return just the list of chats? and when a user needs to read the messages inside the chat, the query is simpler just retrieving all messages that have a `chatId`.